sys/Makefile.include

```diff
@@ -111,3 +111,20 @@ endif
 ifneq (,$(filter clif, $(USEMODULE)))
   INCLUDES += -I$(RIOTBASE)/sys/clif/include
 endif
+
+# zptr can use pointer compression of four byte aligned pointers, by right-
+# shifting them by two and stuffing in an uint16_t.
+# That only works on 32bit platforms, if total RAM size is <=2**18 and the
+# start address is known.
+ifneq (,$(filter zptr,$(USEMODULE)))
+  ifeq (,$(filter arch_8bit arch_16bit,$(FEATURES_USED)))
+    ifneq (,$(RAM_START_ADDR))
+      ifneq (,$(RAM_LEN))
+        # this handles 0xXXX and xxxK
+        ifeq (1,$(shell echo $$(($(subst K,*1024,$(RAM_LEN)) <= 262144))))
+          CFLAGS += -DZPTR_BASE=$(RAM_START_ADDR)
+        endif
+      endif
+    endif
+  endif
+endif
```

Comment:
Is there a better place for this logic? It seems a bit odd to add this to the global `sys` Makefile.include.

Reply:
any suggestion? The "module" consists of just a single header...

Reply:
Not sure how clean it is in terms of build system, but you can add a `sys/zptr` directory with a `Makefile` and a `Makefile.include` (and the `DIRS+=zptr` in `sys/Makefile.dep`). I think that should work fine and it puts the logic in a place where people (IMHO) expect it).

Reply:
Or maybe add `sys/zptr/Makefile.include` and include it here? Why Makefile.dep and a regular Makefile?

Reply:
> add sys/zptr/Makefile.include and include it here? 

did this. Looks cleaner.

Reply:
Looks good to me!